src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs

```diff
@@ -162,9 +162,17 @@ internal class ShapeAccessor : DelegateAccessor
                 {
                     return shape.Items;
                 }
+
                 // Resolve Model.Content.MyNamedPart
                 // Resolve Model.Content.MyType__MyField OR Resolve Model.Content.MyType-MyField
-                return shape.Named(n.Replace("__", "-"));
+                var namedShaped = shape.Named(n.Replace("__", "-"));
+                if (namedShaped != null)
+                {
+                    return namedShaped;
+                }
+
+                // Resolve Model.Content.MyType-MyField-FieldName_Display__DisplayOption
+                return shape.Named(n);
             }
```

Comment:
@jtkech this is still a wip but if you have any time to review here it'd be appreciated.

I had to fix this to get `Model.Content[BlogPost-Tags-TaxonomyField_Display__TagsDisplayText]` to return the shape when I added a display mode for tags, as it creates quite a weird shape name when using display modes.

If you have any thoughts, I'm having trouble getting a template override working for it as well. The templates not crucial, but I think it's highlighting an issue with the display mode shape naming conventions.

`TaxonomyField-TagsDisplayText.Display.liquid` works, but would override for all of those shapes
`BlogPost-TaxonomyField-TagsDisplayText.Display.liquid` doesn't - I think that's probably the correct naming convention, but I'm not sure the shape table is handling the extra display mode options correctly (too many - getting swapped for __ or vice versa).

Reply:
Yes the shape name may be set to a `differentiator` if not null or to the `shapeType`. The `differentiator`depends on the type / part / field and most of the time it only contain a `-` separator. The shape type may start with a stereotype, may contain different things depending on the context e.g `ContentPart_`, here it uses the `_` and `__` separators, and it may contain a display mode, here the code.

        protected string GetDisplayShapeType(string shapeType, BuildFieldDisplayContext context)
        {
            var displayMode = context.PartFieldDefinition.DisplayMode();
            return !String.IsNullOrEmpty(displayMode)
                ? shapeType + "_Display__" + displayMode
                : shapeType;
        }

So here `shapeType = TaxonomyField_Display__TagsDisplayText`, and because it becomes different from the `fieldType`, in the `ContentFieldDisplayDriver` we do this

                    result.Differentiator($"{partName}-{fieldName}-{shapeType}");

Which ends up with a shape `Name = BlogPost-Tags-TaxonomyField_Display__TagsDisplayText`.

---

Yes, `shape.Named(n.Replace("__", "-"));` has been added to allow the following in liquid, here tested with the standard display mode, otherwise as you it would not work.

    Test1: {{ Model.Content.Article-Subtitle.Text }}
    Test2: {{ Model.Content.Article__Subtitle.Text }}
    Test3: {{ Model.Content["Article-Subtitle"].Text }}
    Test4: {{ Model.Content["Article__Subtitle"].Text }}

Before we were able to do only this.

    Test1: {{ Model.Content.Article-Subtitle.Text }}
    Test3: {{ Model.Content["Article-Subtitle"].Text }}

Hmm, i think that would be okay, maybe i missed something.

But a binder has also been added at the end of `Shape.cs`, so that in razor you can use.

    Test: @Model.Content.Article__Subtitle.Text // In place of using `Named()`

So, maybe this has been added mainly for razor where a property name can't contain a `-` character. And then the change in liquid allows to use the same field name with a double `__`.

**But here it's not the same usage of the double `__` than for a template name separator.** Maybe we could use another character, but which one? I will try it tomorrow and think about it.

---

About alternates, just tried with an `Article`with a `Subtitle` in `Header` display mode that can be mapped to your case

    displayType = "" i skip the one equal to "_Detail"
    BlogPost = contentType = Article
    BlogPost = partName = Article = contentType for a direct field
    TaxonomyField = fieldType = TextField
    Tags = fieldName = Subtitle
    TaxonomyField_Display__TagsDisplayText = shapeType = TextField_Display__Header

In `ContentFieldDisplayDriver` in our case where `shapeType != fieldType` we add these alternates

    {fieldType}{displayType}__{shapeType}
    {partType}{displayType}__{fieldName}__{shapeType}
    {contentType}{displayType}__{partName}__{fieldName}__{shapeType}
    {contentType}{displayType}__{fieldType}__{shapeType}

which ends up with

    TextField_Display__Header_Detail
    TextField__TextField_Display__Header
    Article__Subtitle__TextField_Display__Header
    Article__Article__Subtitle__TextField_Display__Header
    Article__TextField__TextField_Display__Header

So in your case

    TaxonomyField_Display__TagsDisplayText_Detail
    TaxonomyField__TaxonomyField_Display__TagsDisplayText
    BlogPost__Tags__TaxonomyField_Display__TagsDisplayText
    BlogPost__BlogPost__Tags__TaxonomyField_Display__TagsDisplayText
    BlogPost__TaxonomyField__TaxonomyField_Display__TagsDisplayText

Which results in these files

    TaxonomyField-TagsDisplayText.Display.liquid
    TaxonomyField-TaxonomyField-TagsDisplayText.Display.liquid
    BlogPost-Tags-TaxonomyField-TagsDisplayText.Display.liquid
    BlogPost-BlogPost-Tags-TaxonomyField-TagsDisplayText.Display.liquid
    BlogPost-TaxonomyField-TaxonomyField-TagsDisplayText.Display.liquid

So, `BlogPost-TaxonomyField-TagsDisplayText.Display.liquid` doesn't belong to the above list.

This is because of the duplicate segments, here `partName` = `contentType` for a direct field, and the `shapeType` starts with the `fieldType`. Maybe normal for the `partName`, at least we could check that the `shapeType` starts with the `fieldType`. **Hmm, maybe we could just remove the duplicates or add more alternates without duplicates to not break existing templates**.

So try one of the above alternates. About your change, right now i would suggest to do `shape.Named(n)` first. Maybe we would have also to change the binder at the end of the `Shape.cs`. I will think about it.

Didn't try any liquid alternate yet, maybe nothing works ;) i will try it tomorrow.

Reply:
Thanks @jtkech I was looking for where those alternates where added and can see the issue clearly now.
 >About your change, right now i would suggest to do

I've done this change, but knowing you will think more about the binder.

So back onto the templates and alternates (noting that I refactored to Tags as the display mode, not TagsDisplayText - it made no sense)

I tried your suggestions for the alternates, but they don't work, because they don't match to a shape binding.
I've put a last commit in here, (a hack, and not intended to be the permanent fix, just to highlight what is going on)

Some definitions (to unconfuse myself)
displayType -> Summary or Detail
displayMode -> the special display mode that can be selected

So when using a display mode the `shapeType` becomes `TaxonomyField_Display__Tags`

But the binding for the template is always
`blogpost_display__blogpost__tags__taxonomyfield__tags` - I assume because the .Display is always placed at the second segment.

But because the shapeType actually contains the `_Display` it doesn't get morphed properly when adding the alternates.

So I did a quick hack to add some extra alternates when the `shapeType` contains `_` and `__`

Maybe this is not such a bad hack, and just needs tweaking somewhat. 
Or maybe another character other than `_` or `__` would make these display options less confusing, as you say? But what character indeed?

I think I've fixed this once now for display modes with placement. Maybe this is the last place that needs fixing, so not an issue to fix it similar to I have.

So thank you for all the great info, and I will let you think about it some more, about the best option.
If you want to push something to this pr, please feel free, or if you have better ideas, then on another pr, as you prefer.
This one is still not complete, and will need another demo at least :)